gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapper.java

```diff
@@ -0,0 +1,301 @@
+            return ivxbts.getValue();
+        } else if (ivxbts.getNullFlavor().equals(CsNullFlavor.UNK)) {
+            return CsNullFlavor.UNK.value();
+        }
+
+        return null;
+    }
+
+    private Date getObservationDate(RCMRMT030101UK04ObservationStatement observationStatement) {
+        if (observationStatement.getEffectiveTime() != null) {
+            var center = getTSStringValue(observationStatement.getEffectiveTime().getCenter());
+            var low = getIVXBTSStringValue(observationStatement.getEffectiveTime().getLow());
+            var high = getIVXBTSStringValue(observationStatement.getEffectiveTime().getHigh());
+            var availabilityTimeValue = observationStatement.getAvailabilityTime();
+
+            if (center != null) {
+                return DateFormatUtil.parseToDateTimeType(center).getValue();
+            } else if (low != null && high != null) {
+                return DateFormatUtil.parseToDateTimeType(low).getValue();
+            } else if (low != null) {
+                return DateFormatUtil.parseToDateTimeType(low).getValue();
+            } else if (high != null && availabilityTimeValue.getValue() != null && availabilityTimeValue.getNullFlavor() == null) {
+                return DateFormatUtil.parseToDateTimeType(availabilityTimeValue.getValue()).getValue();
+            }
+        }
+
+        return null;
+    }
+
+    private String highValueToNotes(RCMRMT030101UK04ObservationStatement observationStatement) {
+        if (observationStatement.getEffectiveTime() != null) {
+            var low = getIVXBTSStringValue(observationStatement.getEffectiveTime().getLow());
+            var high = getIVXBTSStringValue(observationStatement.getEffectiveTime().getHigh());
+            var availabilityTimeValue = observationStatement.getAvailabilityTime();
+
+            if (low != null && high != null) {
+                return END_DATE_PREFIX + high;
+            } else if (high != null && availabilityTimeValue.getValue() != null) {
+                return END_DATE_PREFIX + high;
+            } else if (high != null && availabilityTimeValue.getNullFlavor() != null) {
+                return END_DATE_PREFIX + high;
+            }
+        }
+
+        return null;
+    }
+
+    private Annotation buildNote(RCMRMT030101UK04ObservationStatement observationStatement) {
+        var pertinentText = getPertinentAnnotation(observationStatement);
+        var dateText = highValueToNotes(observationStatement);
+
+        if (pertinentText != null && dateText != null) {
+            var noteStringType = new StringType(pertinentText + StringUtils.SPACE + dateText);
+            return new Annotation(noteStringType);
+        } else if (pertinentText != null) {
+            var noteStringType = new StringType(pertinentText);
+            return new Annotation(noteStringType);
+        } else if (dateText != null) {
+            var noteStringType = new StringType(dateText);
```

Comment:
Minor - but rather than creating the extra variables - we could just do: `return new Annotation(new StringType(dateText))`